xdem/spstats.py

```diff
@@ -9,19 +9,92 @@
 import random
 import warnings
 from functools import partial
-from typing import Callable, Union
+from typing import Callable, Union, Optional
 
 import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
 from scipy import integrate
 from scipy.optimize import curve_fit
+from skimage.draw import disk
+from scipy.interpolate import RegularGridInterpolator, LinearNDInterpolator, griddata
+from xdem.spatial_tools import nmad
 
 with warnings.catch_warnings():
     warnings.filterwarnings("ignore", category=DeprecationWarning)
     import skgstat as skg
     from skgstat import models
 
+def robust_interp_nd_binning(df: pd.DataFrame, var_names: list[str], statistic : Union[str, Callable, None] = nmad, min_count: Optional[int] = 100) -> Callable:
```

Comment:
The callable type should have type hints. 

For `statistic`: I guess this would be `Callable[[np.ndarray], float]`?

For the return: I guess this may take an arbitrary (read: not easy to predict at definition time) amount of arguments, so the type should be something along `Callable[..., np.ndarray]`, right?

Reply:
Thanks for nothign this!
For the return, it's a tuple of any number of arrays, how can we write this? `Callable[tuple[..., np.ndarray]]` ?

Reply:
If it's always a tuple of ndarrays, it should be:
```python
Callable[[tuple[np.ndarray, ...]], np.ndarray]
```
The `Callable` always takes a list as the first (indexer/argument??), whereby it should be a tuple of arbitrary length containing np.ndarrays: `tuple[np.ndarray, ...]`. The next (indexer/argument) shows the return type: a `np.ndarray`.

Reply:
Perfect, thanks a lot!